CMakeLists.txt

```diff
@@ -890,6 +890,13 @@ if( APPLE AND CMAKE_OSX_ARCHITECTURES )
     else()
         set(processor_name ${CMAKE_OSX_ARCHITECTURES})
     endif()
+elseif(CMAKE_CXX_COMPILER_ARCHITECTURE_ID)
+    # When targeting Windows arm64 CMAKE_SYSTEM_PROCESSOR will incorrectly 
+    # return AMD64.
+    # See: https://gitlab.kitware.com/cmake/cmake/-/issues/15170
+    # We assume that when builing for Windows arm64 that we are using MSVC
```

Comment:
It has been recommended that we compile using clangcl.exe not cl.exe as the code runs 10% faster (well the ASTC encoder at least). It is an additional MSVC component that can be downloaded and installed. Do you have any idea how to do that on Actions?

Once it is installed it can be selected with "-T ClangCL" on the CMake configure line.

This code needs to work with that compiler too.  Hopefully since it is an MSVC component, CMAKE_CXX_COMPILER_ARCHITECTURE_ID will still be valid.

Reply:
I love the clang-cl toolset! It's installed by default